translate/filters/checks.py

```diff
@@ -55,14 +55,16 @@
         %(                          # initial %
         (?P<boost_ord>\d+)%         # boost::format style variable order, like %1%
         |
+        \|?                         # boost::format allows to enclose params in pipes, %|spec|
```

Comment:
@unho [there was actually a test for it](https://github.com/translate/translate/blob/master/translate/filters/test_checks.py#L629):
```python
assert passes(stdchecker.printf, "(x,y) = (%|+5|,%|+5|)", "(x,y) = (%|+5|,%|+5|)")`
```
 although that test tested nothing because pipe-brackets were not incorporated in regex, so according to current version of checker this string doesn't have any params. After I implemented a rule "no lone percent signs" this test started to rightfully fail that's why I had to add pipe brackets. So that's why this can't be separated into two PRs. I just added test for this feature: https://github.com/asyschikov/translate/commit/bbd102e5a0e6c8b639c3c26882449db61cc7a437